packages/fxa-settings/src/components/Settings/FlowRecoveryKeyConfirmPwd/index.test.tsx

```diff
@@ -0,0 +1,124 @@
+
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import { act, fireEvent, screen, waitFor } from '@testing-library/react';
+import { logViewEvent } from '../../../lib/metrics';
+import FlowRecoveryKeyConfirmPwd from './';
+import {
+  mockAppContext,
+  MOCK_ACCOUNT,
+  renderWithRouter,
+} from '../../../models/mocks';
+import { Account, AppContext } from '../../../models';
+import { typeByLabelText } from '../../../lib/test-utils';
+
+const localizedBackButtonTitle = 'Back to settings';
+const localizedPageTitle = 'Account Recovery Key';
+const navigateForward = jest.fn();
+const navigateBackward = jest.fn();
+const viewName = 'settings.account-recovery';
+
+jest.mock('../../../lib/metrics', () => ({
+  usePageViewEvent: jest.fn(),
+  logViewEvent: jest.fn(),
+}));
+
+jest.mock('base32-encode', () =>
+  jest.fn().mockReturnValue('00000000000000000000000000000000')
+);
+
+const setFormattedRecoveryKey = jest.fn();
+
+const accountWithSuccess = {
+  ...MOCK_ACCOUNT,
+  createRecoveryKey: jest.fn().mockResolvedValue(new Uint8Array(20)),
+} as unknown as Account;
+
+const renderFlowPage = (account: Account) => {
+  window.URL.createObjectURL = jest.fn();
+  renderWithRouter(
+    <AppContext.Provider value={mockAppContext({ account })}>
+      <FlowRecoveryKeyConfirmPwd
+        {...{
+          localizedBackButtonTitle,
+          localizedPageTitle,
+          navigateForward,
+          navigateBackward,
+          setFormattedRecoveryKey,
+          viewName,
+        }}
+      />
+    </AppContext.Provider>
+  );
+};
+
+describe('FlowRecoveryKeyConfirmPwd', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders as expected', async () => {
+    renderFlowPage(accountWithSuccess);
+
+    screen.getByRole('heading', {
+      name: 'Enter your password again to get started',
+    });
+    screen.getByLabelText('Enter your password');
+    screen.getByRole('button', { name: 'Create account recovery key' });
+  });
+
+  it('disables the submit button when the input is empty', async () => {
+    renderFlowPage(accountWithSuccess);
+    const createRecoveryKeyButton = screen.getByRole('button', {
+      name: 'Create account recovery key',
+    });
+    const passwordInput = screen.getByLabelText('Enter your password');
+    expect(createRecoveryKeyButton).toHaveAttribute('disabled');
+    // input a password to enable form submission
+    fireEvent.input(passwordInput, {
+      target: { value: 'anypassword' },
+    });
+    await waitFor(() =>
+      expect(createRecoveryKeyButton).not.toHaveAttribute('disabled')
+    );
+    // clear password input disables the submit button
+    fireEvent.input(passwordInput, {
+      target: { value: '' },
+    });
+    await waitFor(() =>
+      expect(createRecoveryKeyButton).toHaveAttribute('disabled')
+    );
+  });
+
+  it('emits the expected metrics when the form is submitted with success', async () => {
```

Comment:
Can we get a test for a failed submission as well and maybe check for the appropriate error messages?

When I run `CI=yes yarn test --coverage --verbose`, I see this about this file: `coverage threshold for branches (50%) not met: 40%`